### ISSUE TITLE
profiles: assigning nil to result strings crashes osrm-extract

### ISSUE THREAD

**Issue by emiltin** [Project-OSRM/osrm-backend#3340]
ie this crashes:

```lua
result.ref = nil
```

name, ref and pronunciation fields all have this behaviour. allowing asignment of nil should just be trreated as an empty string, as this simplifies profiles and is safer.

**Comment by emiltin** [comment]
assigning nil to boolean fields result.roundabout also causes a crash. it would be better to treat nil as false.

**Comment by daniel-j-h** [comment]
Here's my understanding: you assign `nil` in Lua - Luabind converts this to a `nullptr` in C++-land.

And assigning a `std::string` a `nullptr` is Undefined Behavior.

**Comment by TheMarex** [comment]
> as this simplifies profiles

Can you elaborate on this?

> and is safer.

I don't think this is a safety concern. I expect that luabind just throws an exception, right?

@daniel-j-h what is your feeling w.r.t to this issue? Currently it seems like expected behavior to me.

**Comment by emiltin** [comment]
by simplifying i mean you could safely do e.g:
```lua
result.name = way:get_value_by_key('name')
```

**Comment by TheMarex** [comment]
Fixed by #3344
